tests/components/cloud/test_onboarding.py

```diff
@@ -0,0 +1,132 @@
+"""Test the onboarding views."""
+
+from http import HTTPStatus
+from typing import Any
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components import onboarding
+from homeassistant.components.cloud import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import register_auth_provider
+from tests.typing import ClientSessionGenerator
+
+
+def mock_onboarding_storage(hass_storage, data):
+    """Mock the onboarding storage."""
+    hass_storage[onboarding.STORAGE_KEY] = {
+        "version": onboarding.STORAGE_VERSION,
+        "data": data,
+    }
+
+
+@pytest.fixture(autouse=True)
+def auth_active(hass: HomeAssistant) -> None:
+    """Ensure auth is always active."""
+    hass.loop.run_until_complete(
+        register_auth_provider(hass, {"type": "homeassistant"})
+    )
```

Comment:
```suggestion
async def auth_active(hass: HomeAssistant) -> None:
    """Ensure auth is always active."""
    await register_auth_provider(hass, {"type": "homeassistant"})
```

Reply:
O wow, that's some really ancient code. I'll clean it up also in the onboarding tests 👍